sdk/core/Azure.Core/src/DelayStrategy.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using Azure.Core.Pipeline;
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// Strategy to control delay behavior.
+    /// </summary>
+    public abstract class DelayStrategy
```

Comment:
Curious: Have we given any consideration to whether/how these would apply to RetriableStream if set?

Reply:
Hmm, it looks like the only overlap would be in the maxRetries as RetriableStream doesn't do any kind of backoff. It looks like all libraries that use RetriableStream hardcode the maxRetries so I don't think we'd need to do anything special.

Reply:
Yeah, I noticed it was pretty bare-bones in RetriableStream - I guess the question would be whether if a customer made changes to RetryPolicy configuration details, if they would have the expectation that those would be applied in a stream-retry case as well.  RetriableStream has a small enough footprint that I wouldn't imagine it has come up yet, but it might be something we want to think about as part of the effort to move it out of shared source when the time for that comes.  Thanks!

Reply:
At least for the current implementations, the retriable stream maxRetries is independent of the user's RetryOptions, so this would already be an existing gap if we end up deciding it should respect the RetryOptions. My worry about correlating them would be if you could somehow get in a situation where N^2 retries are being applied instead of N*M where M is the hard-coded constant. I'd need to dig into the details more to see if this is a concern.

Reply:
Oh, interesting.  I'm looking into using RetriableStream for ACR, and was planning to pass in the max retries value from RetryOptions.  I'll have a quick look to see if it looks like that would be an issue...